_includes/specials/taproot/en/09-ptlcs.md

```diff
@@ -0,0 +1,153 @@
+that she gives to Bob; Bob can use the same point with a public key
+he chooses and creates an adaptor he gives to Carol.  Carol reveals the
+scalar by transforming Bob's adaptor into a valid signature, claiming
+Bob's coins.  Bob recovers the scalar from the valid signature, allowing
+him to transform Alice's adaptor into its own valid signature, claiming
+her coins.
+
+This solves the linkability problem against onchain surveillance because
+all anyone looking at the block chain sees are a bunch of valid
+signatures for distinct public keys.  Third parties can't know that
+adaptors were used, much less what scalar those adaptors were based upon.
+
+However, the procedure above doesn't prevent surveilance nodes who participate
+in the routing from linking together the payments.  If all the payments
+are based on the same scalar, then all the payments are just as linked
+as if they used a hashlock and preimage.  This can be fixed by each
+routing node choosing their own scalar and then removing its
+corresponding point as the payment passes through its node.  Let's
+revise our example:
+
+As before, Carol gives Alice the point for her scalar, but this time
+Alice also requests a point from Bob.  Alice constructs the adaptor she
+gives Bob using the aggregation of both Carol's point and Bob's point.
+Bob knows his point, so he's able to subtract that out from the adaptor
+he receives from Alice.  Using the resultant EC point (which Bob doesn't
+know is now just the point Alice originally received from Carol), Bob
+constructs the adaptor he gives to Carol.  Carol knows the scalar for
+that final point and so converts Bob's adaptor into a valid signature.
+As before, Bob recovers Carol's scalar from her signature and uses it
+and his own scalar to convert Alice's adaptor into a valid signature.
+
+In the two hops in this path, Alice→Bob and Bob→Carol, two different
+EC points and scalars were used, eliminating linkability.  We can extend
+this to the longer path we examined when considering HTLCs and see how
+this improves privacy:
+
+![Illustration of PTLC lack of linkability problem](/img/posts/2021-07-ln-linkability2.dot.png)
+
+As mentioned last week, schnorr signatures make it easy to compose
+adaptor signatures with multisignatures.  This allows us to reduce our
+onchain scripts to:
+
+```hack
+pk($bob_with_alice_adaptor) || (pk($alice) && older(10))
+pk($carol_with_bob_adaptor) || (pk($bob)   && older(5) )
+```
+
+With taproot, the left branch can become a keypath and the right branch
+can become a tapleaf.  If the payment routes successfully, Bob and Carol
+can respectively settle their parts onchain without further cooperation
+from their counterparties, making this routed payment indistinguishable
+from single-sig payments, normal multisignature payments, and
+cooperatively resolved contracts.  It also minimizes the use of block
+space.  Of course, if one of the refund conditions needs to be executed,
+that's still an option.
```

Comment:
I think in that case, the refund transaction looks just like a scriptpath spend where the script is a pubkey + csv. That could be a lightning payment, or many other kinds of payment/transaction.

Is it right to say that even in the unhappy case, there are privacy benefits over the current HTLC construction? Or is that too much of a stretch because there will be other aspects of the transaction that give this away as an LN transaction?

Reply:
Actually, in this paragraph I'm not talking about LN payments specifically but any use of PTLCs.  The transition back to generic PTLCs isn't really clear in the text, sorry.  I'll see if I can fix that and also note that there is still some privacy benefit.

(FWIW, I don't think LN would use the very simple PTLC construction in this post; LN-penalty requires more script stuff and Eltoo uses CLTV in a weird way plus anyprevout, either of which will probably make identification trivial.  But hey, that's why we have a post from ZmnSCPxj next week.  :-)